src/routes/client.ts

```diff
@@ -0,0 +1,408 @@
+import type { FastifyPluginCallback } from 'fastify';
+import type { Snowflake } from 'discord-api-types/v8';
+
+import { WSOpCodes } from '../constants';
+import { Track, TrackDataInput } from '../core/track';
+
+const user: FastifyPluginCallback = async (fastify): Promise<void> => {
+	fastify.post<{ Params: { clientId: Snowflake; guildId: Snowflake }; Body: { channelId?: Snowflake } }>(
+		'/:clientId/:guildId/subscription',
+		{
+			schema: {
+				description: 'Creates a new subscription in the specified guild and channel for the specified client',
+				params: {
+					type: 'object',
+					properties: {
+						clientId: {
+							description: 'The client id of the bot',
+							type: 'string',
+						},
+						guildId: {
+							description: 'The guild id of the guild to create a subscription in',
+							type: 'string',
+						},
+					},
+				},
+				body: {
+					type: 'object',
+					required: ['channelId'],
+					properties: {
+						channelId: {
+							description: 'The channel id of the voice channel to join',
+							type: 'string',
+						},
+					},
+				},
+				response: {
+					204: {
+						description: 'Successful response either creating or reusing a subscription',
+						type: 'null',
+					},
+					400: {
+						description: 'Failure response by not providing a channel id or having an active WebSocket connection',
+						type: 'object',
+					},
+				},
+			},
+		},
+		async (req, reply) => {
+			const clientId = req.params.clientId;
+			const guildId = req.params.guildId;
```

Comment:
Why not something like this?
```suggestion
			const { clientId, guildId } = req.params;
```